### ISSUE TITLE
[GSoC Project Idea]:  QA/QC for glider data

### ISSUE THREAD

**Issue by ocefpaf** [ioos/gsoc#55]
### Project Description

The [ioos_qc](https://github.com/ioos/ioos_qc) library is the canonical implementation of the [QARTOD manuals for ocean QA/QC best practices](https://ioos.noaa.gov/project/qartod/). This project ideas are to: create QA/QC examples and documentation for its use with underwater glider data and implement any missing QA/QC checks for underwater glider data.

#### About Underwater Gliders

Gliders are a unique and important observing system used to serve a variety of subsurface observing missions. Gliders can monitor water currents, temperature, tagged animals and conditions that reveal effects from storms, impacts on fisheries, and the quality of our water. This information creates a more complete picture of what is happening in the ocean, as well as trends scientists might be able to detect. These versatile vehicles collect information from deep water, as well as at the surface, at lower cost and less risk than ever before. As scientists deploy more gliders, they are revolutionizing how we observe our ocean. These robots propel us closer to that revolution.

### Expected Outcomes

Notebook(s) with glider data QA/QC checks, examples, and new QA/QC functions specific for moving platforms (glider).

### Skills required

Ocean data QA/QC knowledge, basic python, numpy, and scipy.

### Mentor(s)

Filipe Fernandes (ocefpaf), Mathew Biddle (MathewBiddle) and Micah Wengren (mwengren )

### Expected Project Size

350

### What is the difficulty of the project?

Intermediate

**Comment by MathewBiddle** [comment]
xref:
* https://github.com/ioos/ioos_qc/issues/58

**Comment by MathewBiddle** [comment]
I added some more details about gliders and links to the various resources.

**Comment by ocefpaf** [comment]
@srstsavage do you, or anyone at Axiom, wants to be a mentor here?

**Comment by mwengren** [comment]
I hope we get some interested candidates in this project!

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf, I'm Saksham Gupta, a final year student from India, and I'm very interested in working on this project. While I'm currently not experienced in Ocean data QA/QC, I'm eager to acquire new skills and learn. I believe that tackling projects with diverse tech stacks is the best way to enhance one's learning abilities, and I thrive in such environments:) Looking forward to the opportunity to contribute to this project!
Is there any specific tasks that I need to do before sending the proposals? And what all I need to learn to work on this project? It would be much more helpful if you could provide me a brief regarding the same.

**Comment by srstsavage** [comment]
@ocefpaf We're discussing mentorship on this project and ioos_qc contributions more broadly. Does anyone know when the deadline for having a specific mentor's name attached would be? The April 2nd proposal deadline probably?

**Comment by MathewBiddle** [comment]
See https://developers.google.com/open-source/gsoc/timeline

The application period opens March 18th and ends April 2nd. What that means is the project ideas need to be ready to go (descriptions of the project and mentor names) by March 18th, and students apply for the project by April 2nd.

@ocefpaf please correct me if I'm misunderstanding.

**Comment by ocefpaf** [comment]
> @ocefpaf please correct me if I'm misunderstanding.

That is correct. Note we can, and probably should, have multiple mentors for the projects. Even more so for this one b/c it will demanding some advanced domain knowledge on glider QA/QC and coding.

**Comment by ocefpaf** [comment]
> Hi @ocefpaf, I'm Saksham Gupta, a final year student from India, and I'm very interested in working on this project. While I'm currently not experienced in Ocean data QA/QC, I'm eager to acquire new skills and learn. I believe that tackling projects with diverse tech stacks is the best way to enhance one's learning abilities, and I thrive in such environments:) 

Hi @Sakshamgupta90!

> Looking forward to the opportunity to contribute to this project! Is there any specific tasks that I need to do before sending the proposals? And what all I need to learn to work on this project? It would be much more helpful if you could provide me a brief regarding the same.

For now you can take a look at [the QARTOD manuals](https://repository.oceanbestpractices.org/handle/11329/336) and the code implemented here. You can try to fork the repository and run all the tests locally to ensure you have a working dev environment.

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf, thanks a lot, I will go through these manuals and the code. I will also try to setup the code locally. Will let you know if got stuck somewhere:)

**Comment by srstsavage** [comment]
After much internal talk it seems like the mentor for this proposal should be someone with deep glider domain experience/expertise, and we don't currently have a great match at Axiom with both timeline and focus area. We're ready and willing to support the effort in other ways, potentially including implementation of missing QA/QC routines identified by this effort, but the lead mentor should probably be a glider data guru. Also 3/18 has come and gone so this may be moot...

**Comment by ocefpaf** [comment]
>  but the lead mentor should probably be a glider data guru

Agreed. We can try to recruit one. Note that having multiple mentors, even it someone will just look at the code quality and not for its QA/QC correctness, is encouraged. If anyone at Axiom wants to collaborate, please send them our way so we can register them as mentors.

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf @mwengren @MathewBiddle 
Happy to be accepted in the program. Really looking forward to this summer and putting in my best work!

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf @MathewBiddle @mwengren 
I'm eager to get started and ensure a smooth transition into the project, I wanted to reach out to discuss the next steps.
How we'll be communicating throughout the program?

**Comment by ocefpaf** [comment]
@Sakshamgupta90 we will send a message to all participants soon. Stay tuned!

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf,
I was going through all the QARTOD manuals, and majority of the tests are same while doing QA/QC of different oceanographic datas.
So, while creating examples of use with glider data, can I refer to the examples created for other?

**Comment by ocefpaf** [comment]
@Sakshamgupta90 I'm not sure if I understand it correctly but:

1. We should try to provide some data examples for this project, keep an eye on https://github.com/ioos/ioos_qc/issues/58.
2. Anything from the existing tests that can be re-used, is fine to re-use.

**Comment by Sakshamgupta90** [comment]
Thanks alot for the insights.
I will go through the codebase once and let you know if there is any issue. And one more thing, to be familiar with QA/QC data, are there any resources which I should go through other than QARTOD manuals?

**Comment by ocefpaf** [comment]
>  are there any resources which I should go through other than QARTOD manuals?

At this point I believe that the manuals are a good start. We should touch base on which ones though, as there are many! I'll ask the experts and get back to you.

**Comment by Sakshamgupta90** [comment]
Hi @ocefpaf, So I was doing the environment setup and this one command is written over there:
``` conda install --file requirements.txt --file tests/requirements.txt```
But in the test folder there is no requirements.txt file present, so I tried installing the requirements.txt file, so is it fine? Do we need tests/requirements.txt file?

**Comment by ocefpaf** [comment]
The command should be:

```
conda install --file requirements.txt --file requirements-dev.txt
```

Can you please point us to where you found the reference to `tests/requirements.txt` and open an issue at https://github.com/ioos/ioos_qc/issues so we can fix it?

**Comment by Sakshamgupta90** [comment]
It is referenced in docs/source/development.rst.
Sure I will open an issue.

**Comment by Sakshamgupta90** [comment]
Hi @MathewBiddle @ocefpaf, 
I've completed the setup and read through all the provided documentation. However, I would like to dive deeper and familiarize myself further with glider data. Could you please point me to additional resources, examples, or any relevant materials?

Additionally, I was looking for documentation and examples related to QA/QC checks for other resources but haven't been able to find them. Could you share the links to these if available?

**Comment by mwengren** [comment]
Closing to clean up for 2025 GSoC application.
